### PR TITLE
Added iOS Object Detector Objective C Tests

### PR DESCRIPTION
**Pull request by priankakariatyml**
1. Removed Roi apis as object detection does not support rot
2. Fixed bug in check for whether roi is allowed in MPPVIsionTaskRunner.
3. Added Object Detector Objective C Tests

### CODE REVIEW

File: mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m [hunk 1]
```diff
@@ -0,0 +1,703 @@
...
+#import "mediapipe/tasks/ios/test/vision/utils/sources/MPPImage+TestUtils.h"
+#import "mediapipe/tasks/ios/vision/object_detector/sources/MPPObjectDetector.h"
+
+static NSString *const kModelName = @"coco_ssd_mobilenet_v1_1.0_quant_2018_06_29";
+static NSDictionary *const kCatsAndDogsImage = @{@"name" : @"cats_and_dogs", @"type" : @"jpg"};
+static NSDictionary *const kCatsAndDogsRotatedImage =
+    @{@"name" : @"cats_and_dogs_rotated", @"type" : @"jpg"};
+static NSString *const kExpectedErrorDomain = @"com.google.mediapipe.tasks";
+
+#define PixelDifferenceTolerance 5.0f
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m:27]: Let's use proper `const` variables for these.

File: mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+#define ScoreDifferenceTolerance 1e-2f
+
+#define AssertEqualErrors(error, expectedError)                                               \
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m:30]: I think these macros are getting a bit more complicated than what I would think is acceptable for macros. Can we turn this into helper functions?

File: mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m [hunk 3]
```diff
@@ -0,0 +1,703 @@
...
+  MPPObjectDetectorOptions *options = [self objectDetectorOptionsWithModelName:kModelName];
+  options.maxResults = 1;
+
+  MPPObjectDetector *objectDetector = [self objectDetectorWithOptionsSucceeds:options];
+
+  NSArray<MPPDetection *> *detections = @[
+    [[MPPDetection alloc] initWithCategories:@[
+      [[MPPCategory alloc] initWithIndex:-1
+                                   score:0.750000f
+                            categoryName:@"teddy bear"
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m:422]: There is a bug in how we handle image rotation in all of our tasks. We are going through the APIs and fixing this. Can you see if you can port https://github.com/google/mediapipe/commit/e3185e3df05dc0ef347118c18e08b8a9ff5263d4 and verify that we are getting a "cat" here?

File: mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m (hunk 1 above)
Comment (priankakariatyml) [reply mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m]: Fixed

File: mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m (hunk 2 above)
Comment (priankakariatyml) [reply mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m:29]: I have converted them into helper functions which use macros only for comparison of two categories and bounding boxes.

**Pull request merged**